src/dataflow/src/source/s3.rs

```diff
@@ -716,6 +735,7 @@ async fn download_object(
     key: &str,
     compression: Compression,
     source_id: &str,
+    skip_bytes: usize,
 ) -> (DownloadStatus, Option<DownloadMetricUpdate>) {
     let obj = match client
         .get_object(GetObjectRequest {
```

Comment:
If `skip_bytes` is non-zero we can use a [range header](https://rusoto.github.io/rusoto/rusoto_s3/struct.GetObjectRequest.html#structfield.range) here to let S3 do the seeking to the right offset and avoid re-reading the file further down.

I saw this is also mentioned as a limitation of S3 in our docs but it doesn't sound right https://github.com/MaterializeInc/materialize/blob/main/doc/user/layouts/partials/create-source/connector/s3/details.html#L11-L12

Reply:
Good point! I didn't actually realize this was possible if you didn't originally PUT the object chunked.

hmm yeah those docs are meant to demonstrate that we can't `tail` an individual object, that's just wrong.